src/editors/_shared/hooks/DataTypes/types/BooleanType.js

```diff
@@ -5,90 +5,45 @@ import { pythonToBool, boolToPython } from "../../../../../utils/Utils";
 import DataType from "../AbstractDataType";
 
 class BooleanType extends DataType {
-  // Boolean type properties definition
   key = DATA_TYPES.BOOLEAN;
   label = "Boolean";
   default = false;
 
-  editComponent = (props, mode = "row") => {
-    this.label = props.label ?? this.label;
-    let pyValue = this.toString(props.rowData.value).toLowerCase();
-    const editor = {
-      row: () => this.booleanEditComponent(props, pyValue),
-      dialog: () => this.booleanEditComponent(props, pyValue, true)
-    };
-    return editor[mode]();
-  };
+  // boolType does parsing in reverse, since it only
+  // wants to parse values when we're in onlyStrings.
+  // it also uses a non-text input
 
-  /**
-   * Validate Boolean value
-   * @param {*} value
-   * @returns
-   */
-  validate(value) {
-    return new Promise(resolve => {
-      try {
-        const isValid =
-          typeof value === DATA_TYPES.BOOLEAN ||
-          typeof pythonToBool(value) === DATA_TYPES.BOOLEAN;
-
-        resolve({ success: isValid });
-      } catch (e) {
-        resolve({ success: false });
-      }
-    });
+  constructor(opts) {
+    super({ ...opts, textInput: false });
   }
 
-  /**
-   * @override Get boolean default value
-   * @param {*} options
-   * @returns {string | boolean}
-   *  Returns Python Boolean as string if isPythonValue is set to true
-   *  Or js boolean otherwise
-   */
-  getDefault(options = { isPythonValue: false }) {
-    const { isPythonValue } = options;
-    return isPythonValue ? boolToPython(false) : false;
+  parse(value) {
+    if (value === "")
+      return undefined;
+    if (value === "True")
+      return true;
+    if (value === "False")
+      return false;
+    return null;
   }
```

Comment:
In which cases we want to return `undefined` and in which ones we want to return `null`?
With this suggestion it'll always return `true` / `false` / `undefined` accordingly.

```suggestion
  parse(value) {
    const possibleValues = {
      "True": true,
      "False": false
    }
    return possibleValues[value];
  }
```

Reply:
That is not equivalent.